### PR TITLE
[JENKINS-29781] BuildTrigger does not reflect moving around declaring job

### PR DESCRIPTION
**Pull request by olivergondza**
[JENKINS-29781](https://issues.jenkins-ci.org/browse/JENKINS-29781)

### CODE REVIEW

**Comment by olivergondza** [comment]
Can someone confirm that the behaviour from the unittest is the desired one?

File: core/src/main/java/hudson/model/Items.java [hunk 1]
```diff
@@ -242,25 +250,66 @@ public static String getCanonicalName(ItemGroup context, String path) {
...
+    public static String computeRelativeNamesAfterRenaming(String oldFullName, String newFullName, String relativeNames, AbstractItem context) {
+        String contextFullName = context.getFullName();
+        String newGroupName = context.getParent().getFullName();
+        String oldGroupName = newGroupName;
+        if (contextFullName.equals(newFullName)) { // We are renaming the declared job, not the referred one
+            int i = oldFullName.lastIndexOf("/");
+            oldGroupName = i == -1 ? "" : oldFullName.substring(0, i);
+        } else if (contextFullName.startsWith(newFullName + '/')) {
+            // Renaming ItemGroup that contains subject
+            // Will be handled in subsequest call to #onLocationChanged for the project itself
```
Comment (olivergondza) [inline core/src/main/java/hudson/model/Items.java]: I do not like this, though I do not like delegating this to clients as well.

File: core/src/main/java/hudson/model/Items.java [hunk 2]
```diff
@@ -242,25 +250,66 @@ public static String getCanonicalName(ItemGroup context, String path) {
      * @param oldFullName the old full name of the item
      * @param newFullName the new full name of the item
      * @param relativeNames coma separated list of Item relative names
-     * @param context the {link ItemGroup} relative names refer to
-     * @return relative name for the renamed item, based on the same ItemGroup context
+     * @param context the {link AbstractItem} relative names refer to
+     * @return relative name for the renamed item, based on the same AbstractItem context
+     * @since TODO
+     */
+    public static String computeRelativeNamesAfterRenaming(String oldFullName, String newFullName, String relativeNames, AbstractItem context) {
```
Comment (jglick) [inline core/src/main/java/hudson/model/Items.java]: Should add coverage to `ItemsTest` (unit tests).

Comment (olivergondza) [inline core/src/main/java/hudson/model/Items.java]: Right, working on that.

File: core/src/main/java/hudson/model/Items.java [hunk 3]
```diff
@@ -201,7 +205,11 @@ public static String toNameList(Collection<? extends Item> items) {
      * root.
      */
     public static String getCanonicalName(ItemGroup context, String path) {
-        String[] c = context.getFullName().split("/");
+        return getCanonicalName(context.getFullName(), path);
+    }
+
+    public static String getCanonicalName(String groupFullName, String path) {
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Items.java]: `@Nonnull` both?

**Comment by olivergondza** [comment]
Abandoned

**Pull request closed without merging**